pkg/sink/kafka/admin.go

```diff
@@ -39,11 +52,44 @@ func NewSaramaAdminClient(ctx context.Context, config *Options) (ClusterAdminCli
 	if err != nil {
 		return nil, err
 	}
-	return &saramaAdminClient{client: client}, nil
+	return &saramaAdminClient{
+		client:          client,
+		brokerEndpoints: config.BrokerEndpoints,
+		config:          saramaConfig,
+	}, nil
+}
+
+func (a *saramaAdminClient) reset() error {
+	newClient, err := sarama.NewClusterAdmin(a.brokerEndpoints, a.config)
+	if err != nil {
+		return cerror.Trace(err)
+	}
+
+	_ = a.client.Close()
+	a.client = newClient
+
+	return errors.New("retry after reset")
 }
 
-func (a *saramaAdminClient) GetAllBrokers(context.Context) ([]Broker, error) {
-	brokers, _, err := a.client.DescribeCluster()
+func (a *saramaAdminClient) GetAllBrokers(ctx context.Context) ([]Broker, error) {
+	var (
+		brokers []*sarama.Broker
+		err     error
+	)
+	err = retry.Do(ctx, func() error {
```

Comment:
all the retry logic is almost the same, can you refine it to reuse same code?

Reply:
+1, Could the retry logic be extracted to a common place?

Reply:
yes, we can do it, but let us do it when it's not in a hurry.

Reply:
done, we can use high order function to solve this.